modules/core/src/lib/layer.js

```diff
@@ -193,13 +193,10 @@ export default class Layer extends Component {
   // Always unprojects to the viewport's coordinate system
   unproject(xy) {
     const {viewport} = this.context;
-    assert(Array.isArray(xy));
```

Comment:
Many of these asserts can be replaced by simple static type checking. For instance, without even adopting typescript, running the `typescript` checker  and just letting it parse the existing code and jsdoc comment headers, can catch a lot of incorrect parameter passing cases. It does require some massaging of existing code to make it work, I have a proof-of-concept PR in math.gl for that.